server/routes/login.js

```diff
@@ -0,0 +1,41 @@
+const {Chef,Customer} = require('../models/index')
+const express = require('express');
+const router = express.Router();
+const _ = require('lodash');
+const bcrypt = require('bcrypt');
+const Joi = require('joi');
+const jwt = require('jsonwebtoken');
+const config = require('config');
+
+//handle request for /login
+//check user email password and return user, jwt token and user type
+router.post('/', async (req,res)=>{
+    const { error } = validate(req.body);
+    if (error) return res.status(400).send(error.details[0].message);
+    
+    let user = await Chef.findOne({ email:req.body.email });
+    let usertype = "chef";
+    if (!user) {
+        user = await Customer.findOne({ email:req.body.email });
+        usertype = "customer";
+        if (!user) {
+            return res.status(400).send("Invalid email or password");
+        }
+    }
+    const isvalidpassword = await bcrypt.compare(req.body.password, user.password);
```

Comment:
The User model has a method that can accomplish this in a slightly more concise way. It's just a wrapper around bcrypt.compare.

```suggestion
    const isvalidpassword = await user.comparePassword(req.body.password);
```

Reply:
Changed and tested, codde is pushed in new commit